pkg/fleetallocation/controller.go

```diff
@@ -272,6 +273,29 @@ func (c *Controller) allocate(f *stablev1alpha1.Fleet, fam *stablev1alpha1.Fleet
 		return allocation, err
 	}
 
+	// for each node, count how many ready and allocated gameservers exist
+	nodeCounts := countReadyAndAllocatedPerNode(gsList)
+
+	// prioritise by nodes that already have allocated gameservers, and then by ready
+	sort.Slice(gsList, func(i, j int) bool {
```

Comment:
We are allocating just one server here so we dont need to actually sort the list (nlogn), just to find the best one (O(n)).

So we should combine this new logic with the iteration below.

Reply:
Excellent catch. Working on it now.